src/components/modals/FormStepSaveModal.js

```diff
@@ -0,0 +1,115 @@
+ */
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {useHistory} from 'react-router-dom';
+import {FormattedMessage} from 'react-intl';
+import {put, post, destroy} from 'api';
+import {getBEMClassName} from 'utils';
+import Modal from 'components/modals/Modal';
+import ErrorMessage from 'components/ErrorMessage';
+import Body from 'components/Body';
+import Label from 'components/Label';
+import Input from 'components/Input';
+import HelpText from 'components/HelpText';
+import {Toolbar, ToolbarList} from 'components/Toolbar';
+import Button from 'components/Button';
+
+
+const FormStepSaveModal = ({
+    isOpen,
+    closeModal,
+    stepData,
+    saveStepDataUrl,
+    suspendFormUrl,
+    destroySessionUrl,
+}) => {
+  const history = useHistory();
+  const [email, setEmail] = useState("");
+  const [failed, setFailed] = useState(false);
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    let response = await put(saveStepDataUrl, {data: stepData});
+    if (!response.ok) return setFailed(true);
+    response = await post(suspendFormUrl, {email});
+    if (!response.ok) return setFailed(true);
+    try {
+      // Destroy throws an exception if the API is not successful
+      await destroy(destroySessionUrl);
+    } catch (e) {
+      return setFailed(true);
+    }
+    history.push('/');
+  };
+
+  return (
+      <Modal
+        isOpen={isOpen}
+        closeModal={closeModal}
+      >
+        <Body component="form" onSubmit={onSubmit}>
+
+          {
+            failed
+            ? (
+              <ErrorMessage>
+                <FormattedMessage
+                  description="Form save modal error message"
+                  defaultMessage="Saving form failed.  Please try again."
+                />
+              </ErrorMessage>
+            )
+            : null
+          }
+
+          <Body modifiers={['big']}>
+            <FormattedMessage
+              description="Form save modal body text"
+              defaultMessage={"Enter your email address to get an email to resume the form at a later date"}
+            />
+          </Body>
+
+          <div className={getBEMClassName('form-control')}>
+
+            <Label isRequired >
+              <FormattedMessage
+                description="Form save modal email field label"
+                defaultMessage="Your email address" />
+            </Label>
+
+            <Input type="email" value={email} onChange={event => setEmail(event.target.value)} />
+
+            <HelpText>
+              <FormattedMessage
+                description="Form save modal email field help text"
+                defaultMessage="The email address where you will received the email about resuming the form."
+              />
+            </HelpText>
+
+          </div>
+
+          <Toolbar modifiers={['bottom', 'reverse']}>
+            <ToolbarList>
+              <Button type="submit" variant="primary">
```

Comment:
This button should be disabled while the submissions is happening (`put`, `post` and `delete` calls happening) to prevent double-submission. End-users _will_ double click.

Additionally, you want some kind of user-feedback while the backend is processing, so please display a loader after the user submitted.